src/tools/pkcs11-register.c

```diff
@@ -136,7 +136,8 @@ add_module_pkcs11_txt(const char *profile_dir,
 	p[pkcs11_txt_len] = '\0';
 	pkcs11_txt = p;
 
-	if (!strstr(pkcs11_txt, module_path)) {
+	if (!strstr(pkcs11_txt, module_path)
+			&& (exclude_module_path && !strstr(pkcs11_txt, exclude_module_path))) {
```

Comment:
Should it be `&& (!exclude_module_path || !strstr(pkcs11_txt, exclude_module_path))` instead?

Because it seems that currently any invocation with an explicit command line argument will do nothing.

Reply:
Yes, you're right, thanks for the review